apps/workflow-server/gcc-workflow-server-util/src/main/java/com/coremedia/labs/translation/gcc/util/Settings.java

```diff
@@ -0,0 +1,335 @@
+package com.coremedia.labs.translation.gcc.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import edu.umd.cs.findbugs.annotations.UnknownNullness;
+import org.slf4j.Logger;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.lang.invoke.MethodHandles.lookup;
+import static java.util.Objects.requireNonNull;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Represents settings for GlobalLink, typically merged from different sources.
+ * <p>
+ * This record provides a convenient way to combine configuration data from
+ * multiple sources while maintaining type safety and enabling deep merging
+ * of nested maps. Later sources take precedence over earlier ones.
+ *
+ * @param properties merged properties from all sources
+ * @since 2506.0.0-1
+ */
+public record Settings(@NonNull Map<String, Object> properties) {
+  /**
+   * The logger for this class.
+   */
+  @NonNull
+  private static final Logger LOG = getLogger(lookup().lookupClass());
+
+  /**
+   * An empty settings instance with no properties.
+   */
+  @NonNull
+  public static final Settings EMPTY = new Settings(Map.of());
+
+  /**
+   * Defines the global configuration path.
+   * <p>
+   * The path may denote a (settings) document as well as a folder that contains
+   * settings documents to be respected for the GlobalLink settings.
+   */
+  @VisibleForTesting
+  public static final String GLOBAL_CONFIGURATION_PATH = "/Settings/Options/Settings/Translation Services";
+
+  /**
+   * Defines the site-specific configuration path.
+   * <p>
+   * If a GlobalLink parameter should be different in a specific site,
+   * then the path either denotes a site-specific (settings) document or a
+   * folder that may contain settings documents to be used to determine the
+   * GlobalLink settings.
+   */
+  @VisibleForTesting
+  public static final String SITE_CONFIGURATION_PATH = "Options/Settings/Translation Services";
```

Comment:
Remove VisibleForTesting. The constant is used in GlobalLinkAction, which is production code in a different package.